odl/tomo/data/uncompr_bin.py

```diff
@@ -0,0 +1,716 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Utilities and class for reading uncompressed binary files with header."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import int, object, open, str
+
+from collections import OrderedDict
+import csv
+import numpy as np
+import struct
+
+
+__all__ = ('FileReaderRawBinaryWithHeader',
+           'FileWriterRawBinaryWithHeader',
+           'header_fields_from_table')
+
+
+def _fields_from_table(spec_table, id_key):
+    """Read the specification and return a list of fields.
+
+    The given specification is assumed to be in
+    `reST grid table format
+    <http://docutils.sourceforge.net/docs/user/rst/quickref.html#tables>`_.
+
+    Parameters
+    ----------
+    spec_table : str
+        Specification given as a string containing a definition table.
+    id_key : str
+        Dictionary key (= column header) for the ID labels in ``spec``.
+
+    Returns
+    -------
+    fields : tuple of dicts
+        Field list of the specification with combined multi-line entries.
+        Each field corresponds to one (multi-)line of the spec.
+    """
+    # Reformat the table, throwing away lines not starting with '|' or
+    # containing '...'.
+    spec_lines = [line[1:-1].rstrip() for line in spec_table.splitlines()
+                  if line.startswith('|') and '...' not in line]
```

Comment:
It's not 100 % watertight, I agree. The best thing is to remove that "feature" and to just squeeze the table.